mighty/mighty_agents/base_agent.py

```diff
@@ -603,8 +606,23 @@ def run(  # noqa: PLR0915
                 metrics["episode_reward"] = episode_reward
 
                 action, log_prob = self.step(curr_s, metrics)
-                next_s, reward, terminated, truncated, _ = self.env.step(action)  # type: ignore
-                dones = np.logical_or(terminated, truncated)
+                # 1) step the env as usual
+                next_s, reward, terminated, truncated, infos = self.env.step(action)
+
+                # 2) decide which samples are true “done”
+                replay_dones = terminated          # physics‐failure only
```

Comment:
Comment here is env-specific. Also inconsistent: dones are always overwritten to real termination regardless of what the flag says.

Reply:
Make this default